dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -237,14 +240,57 @@ std::wstring Microsoft::Windows::AppNotifications::Helpers::GetDisplayNameBasedO
     return displayName;
 }
 
-// Placeholder
-HRESULT RetrieveAssetsFromProcess(_Out_ Microsoft::Windows::AppNotifications::Helpers::AppNotificationAssets& /*assets*/)
+inline wil::unique_hicon RetrieveIconFromProcess()
 {
-    // THROW_HR_IF_MSG(E_UNEXPECTED, VerifyIconFileExtension(iconFilePath));
+    std::wstring processPath{};
+    THROW_IF_FAILED(wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, processPath));
 
-    return E_NOTIMPL;
+    // Extract the small icon from the first .ico, if failed extract the large icon.
```

Comment:
Add a comment: Small icon is sufficient for Toast icons since higher pixel count doesn't impact the visual in any substantial way